TockLibrary.mk

```diff
@@ -91,28 +92,35 @@ endef
 $(foreach hdrdir,$($(LIBNAME)_SRCS_DIRS),$(eval $(call LIB_HEADER_INCLUDES,$(hdrdir))))
 
 # Rules to generate libraries for a given Architecture
-# These will be used to create the different architecture versions of LibNRFSerialization
+# These will be used to create the different architecture versions of libraries.
+#
+# - `$(LIBNAME)_FOLDER`: The folder name of the library. E.g., `libtock-sync`.
+#
 # Argument $(1) is the Architecture (e.g. cortex-m0) to build for
 define LIB_RULES
 
-$$($(LIBNAME)_BUILDDIR)/$(1):
+$$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER):
 	$$(TRACE_DIR)
 	$$(Q)mkdir -p $$@
 
-$$($(LIBNAME)_BUILDDIR)/$(1)/%.o: %.c | $$($(LIBNAME)_BUILDDIR)/$(1) newlib-$$(NEWLIB_VERSION_$(1))
+$$($(LIBNAME)_BUILDDIR)/$(1)/%.o: %.c | $$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER) newlib-$$(NEWLIB_VERSION_$(1))
 	$$(TRACE_CC)
+	$$(Q)mkdir -p $$(dir $$@)
```

Comment:
You should be able to just remove this, the `| $$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER)` of the rule above makes sure the directory is built first [but only cares about its existence, and ignores its timestamp, so it doesn't trigger a rebuild if the directory is "newer" which will always be true since files in it are modified after; that's what the `|` "order only" separator for rules does]

I quick 'n dirty tested:
```diff
$ git diff
diff --git a/TockLibrary.mk b/TockLibrary.mk
index d8093c1..ebed75f 100644
--- a/TockLibrary.mk
+++ b/TockLibrary.mk
@@ -105,13 +105,13 @@ $$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER):

 $$($(LIBNAME)_BUILDDIR)/$(1)/%.o: %.c | $$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER) newlib-$$(NEWLIB_VERSION_$(1))
        $$(TRACE_CC)
-       $$(Q)mkdir -p $$(dir $$@)
+       $$(Q)echo "FAKE mkdir -p $$(dir $$@)"
        $$(Q)$$(TOOLCHAIN_$(1))$$(CC_$(1)) $$(CFLAGS) $$(CFLAGS_$(1)) $$(CPPFLAGS) $$(CPPFLAGS_$(1)) $$(CPPFLAGS_$(LIBNAME)) -MF"$$(@:.o=.d)" -MG -MM -MP -MT"$$(@:.o=.d)@" -MT"$$@" "$$<"
        $$(Q)$$(TOOLCHAIN_$(1))$$(CC_$(1)) $$(CFLAGS) $$(CFLAGS_$(1)) $$(CPPFLAGS) $$(CPPFLAGS_$(1)) $$(CPPFLAGS_$(LIBNAME)) -c -o $$@ $$<

 $$($(LIBNAME)_BUILDDIR)/$(1)/%.o: %.S | $$($(LIBNAME)_BUILDDIR)/$(1) newlib-$$(NEWLIB_VERSION_$(1))
        $$(TRACE_AS)
-       $$(Q)mkdir -p $$(dir $$@)
+       $$(Q)echo "FAKE mkdir -p $$(dir $$@)"
        $$(Q)$$(TOOLCHAIN_$(1))$$(AS) $$(ASFLAGS) $$(CPPFLAGS) $$(CPPFLAGS_$(1)) -c -o $$@ $$<

 $(LIBNAME)_OBJS_$(1) += $$(patsubst $$($(LIBNAME)_DIR)/%.s,$$($(LIBNAME)_BUILDDIR)/$(1)/$$($(LIBNAME)_FOLDER)/%.o,$$(filter %.s, $$($(LIBNAME)_SRCS)))
```

which seemed to do the right things:
```
riscv64-unknown-elf-ranlib ../simple-ble/build/rv32imac/simple-ble.a
mkdir -p ../simple-ble/build/rv32imc/simple-ble
echo "FAKE mkdir -p ../simple-ble/build/rv32imc/simple-ble/"
FAKE mkdir -p ../simple-ble/build/rv32imc/simple-ble/
```
(that first, real `mkdir` is coming from the rule on line 102)

Reply:
No, that only creates `../simple-ble/build/rv32imc/simple-ble`, we need something like `../libtock-sync/build/cortex-m0/libtock-sync/crypto`. (Crypto being the difference)

Relevant output:

```
  CC        ../../libtock/storage/syscalls/kv_syscalls.c
  CC        ../../libtock/storage/syscalls/nonvolatile_storage_syscalls.c
  CC        ../../libtock/storage/syscalls/sdcard_syscalls.c
  AR        ../../libtock/build/rv32imc/libtock.a
 DIR        ../../libtock-sync/build/cortex-m0/libtock-sync
  CC        ../../libtock-sync/crypto/hmac.c
<built-in>: fatal error: opening dependency file ../../libtock-sync/build/cortex-m0/libtock-sync/crypto/hmac.d: No such file or directory
compilation terminated.
make: *** [../../libtock-sync/build/cortex-m0/libtock-sync/crypto/hmac.o] Error 1
```